src/engine.ts

```diff
@@ -529,6 +593,12 @@ export function initializePolyfill(updateCallback: () => void) {
           ) {
             getOrCreateInstance(child).update(currentState);
           }
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          const shadow = (node as any)[SHADOW_SYMBOL];
+          if (shadow) {
+            updateShadowHostState(shadow, currentState);
```

Comment:
This shouldn't be needed here anymore: the state updater should be called by the `update` call

Reply:
Great catch! I forgot to drop this when moving the function to assign the `stateProvider`.